docs/DeveloperGuide.md

```diff
@@ -23,12 +131,274 @@
+ !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+```
+
+#### 2. Register New User
+##### Test case 2.0: Correctly registers a user
+Registers a new user, when the formatting provided is correct
+
+**Test case:**
+
+Enter choice: 2
+
+Enter email address: `mary@gmail.com`
+
+Enter password: `1234`
+
+Expected:
+```` Hello! I'm MojoHR
+  _________   _____   _____   _____
+ |  _   _  | |  _  | |_   _| |  _  |
+ | | | | | | | | | |   | |   | | | |
+ | | | | | | | |_| |  _| |   | |_| |
+ |_| |_| |_| |_____| |___|   |_____|
+ What can I do for you?
+ > Use the keyword "LIST (type) " to print the emails by types
+ > Use the keyword "READ (index) " to open the selected email
+ > Use the keyword "COMPOSE " to create a draft email
+ > Use the keyword "SEND (index of draft) " to send email in the draft folder
+ > Use the keyword "DELETE (index) " to delete the selected email
+ > Use the keyword "FIND (keyword) " to find the  email by keywords
+ > Use the keyword "ARCHIVE (index) " to move the selected email to the archive folder
+ > Use the keyword "TAG (index)" to select the email for tagging labels
+ > Use the keyword "RESET" to reset the your account password
+ > Use the keyword "HELP" to print the menu
+ > Use the keyword "BYE" to exit
+ ____________________________________________________________
+ Enter Command: 
+````
+
+##### Test case 2.1: Incorrect Registration Format provided by a User
+Error message shown, when the formatting provided is not correct
+
+**Test case**
+
+Enter choice: `2`
+
+Enter email address: `hi`
+
+Enter password: `1234`
+
+Expected
+````
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+Error Message: 
+Please enter a valid email address! 
+Email address must have "@" and cannot have empty string in front or behind
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+````
+
+#### 3. Exits Application
+##### Test case 3.0: Allow the user to exit the system
+When the correct choice format is chosen, user will be able to exit the system
+
+**Test case**
+
+Enter choice: 3
+
+Expected:
+```
+Logging off... Hope to see you again in MojoHr!
+```
+
+#### Test case 3.1: Incorrect Choice Format
+When an incorrect format of choice is shown, error message will be displayed
+
+Enter choice: `three`
+
+Expected:
+```
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+Error Message: 
+You need to enter an integer! Please try again!
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+```
+
+
+### Finding emails by keyword
+Print all emails that contains the keyword, and if there is no email contains the keyword, outputs `No matching emails found.`
+
+Test case: `find subject 10`
+
+Expected: 
+````
+1. [Deleted][UNREAD]
+|| Subject: This is subject 10
+|| From: 21312@gmail.com --> To: [12312@gmail.com]
+|| Time: 2021-02-20T15:30:00
+|| Tags: []
+````
+
+Test case: `find school`
+
+Expected: `No matching emails found.`
+
+
+##Resetting the password
```

Comment:
![Screenshot 2021-04-03 at 12 39 06 PM](https://user-images.githubusercontent.com/65282395/113468590-96fc4180-9479-11eb-8cef-9a5486dc95f2.png)

Same as above, should there be a space between ## and Resetting?